src/webapp/reports/mal-data-approval/data-approval-list/hooks/useDataApprovalFilters.tsx

```diff
@@ -0,0 +1,142 @@
+import _ from "lodash";
+import { useCallback, useEffect, useMemo, useState } from "react";
+import { DropdownProps, MultipleDropdownProps } from "@eyeseetea/d2-ui-components";
+import { useAppContext } from "../../../../contexts/app-context";
+import { DataSetsFilter } from "../Filters";
+import { OrgUnitsFilterButtonProps } from "../../../../components/org-units-filter/OrgUnitsFilterButton";
+import { getOrgUnitsFromId, getRootIds } from "../../../../../domain/common/entities/OrgUnit";
+import { OrgUnitWithChildren } from "../../../../../domain/reports/mal-data-approval/entities/OrgUnitWithChildren";
+
+type DataApprovalFilterProps = {
+    values: DataSetsFilter;
+    onChange: (filter: DataSetsFilter) => void;
+};
+
+type DataApprovalFilterState = {
+    filterValues: DataSetsFilter;
+    rootIds: string[];
+    selectableIds: string[];
+    applyFilters: () => void;
+    clearFilters: () => void;
+    setFilterValues: {
+        dataSetId: SingleDropdownHandler;
+        orgUnitPaths: OrgUnitsFilterButtonProps["setSelected"];
+        periods: DropdownHandler;
+        completionStatus: SingleDropdownHandler;
+        approvalStatus: SingleDropdownHandler;
+    };
+};
+
+export function useDataApprovalFilters(filterProps: DataApprovalFilterProps): DataApprovalFilterState {
+    const { config, compositionRoot } = useAppContext();
+    const { values: filter, onChange } = filterProps;
+    const { orgUnitPaths } = filter;
+
+    const [filterValues, setFilterValues] = useState(emptyApprovalFilter);
+    const [orgUnits, setOrgUnits] = useState<OrgUnitWithChildren[]>([]);
+
+    useEffect(() => {
+        compositionRoot.malDataApproval.getOrgUnitsWithChildren().then(setOrgUnits);
+    }, [compositionRoot.malDataApproval]);
+
+    const dataSetOrgUnits = getOrgUnitsFromId(config.orgUnits, orgUnits);
+    const selectableOUs = _.union(
+        orgUnits.filter(org => org.level < 3),
```

Comment:
can we reuse the `countryLevel` variable here? If we need to update this in the future we can do it in only one place